.travis.yml

```diff
@@ -19,19 +19,32 @@ env:
         # Make sure beaver is in the PATH
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
     matrix:
-        - DMD=1.081.1 DIST=xenial F=production AFTER_SCRIPT=1
+        - DMD=1.081.1 DIST=xenial F=production
         - DMD=1.081.1 DIST=xenial F=devel
         - DMD=2.070.2.s12 DIST=xenial F=production
         - DMD=2.070.2.s12 DIST=xenial F=devel
         - DMD=2.071.2.s1 DIST=xenial F=production
         - DMD=2.071.2.s1 DIST=xenial F=devel
 
+# Don't build tags already converted to D2
+if: NOT tag =~ \+d2$
+
 install: beaver dlang install
 
-script: BEAVER_DOCKER_VARS="F AFTER_SCRIPT" beaver dlang make
+script: beaver dlang make
+
+after_success: beaver dlang codecov
 
-after_success:
-    - test "$AFTER_SCRIPT" = "1" && beaver run ci/codecov.sh
-    - test "$AFTER_SCRIPT" = "1" &&
-        BEAVER_DOCKER_VARS="OCEAN_D2_USER OCEAN_D2_PASS"
-        beaver run ci/d2tags.sh
+jobs:
+    include:
+        - stage: D2 Release
+          # We need to include the exclusion of D2 tags because this "if"
+          # replaces the global one
+          if: tag IS present AND NOT tag =~ \+d2$
+          # We override it, otherwise is inherited from the first matrix row,
+          # we need DIST and DMD for `beaver dlang install`
+          env: DIST=xenial DMD=2.070.2.s12
```

Comment:
Remind me, why did it matter for `DIST` and `DMD` to be different from first matrix row? `make d2conv` doesn't depend on D compiler in any way.

Reply:
IIRC if you don't "fix" the matrix, the job will be run for all the matrix rows. I just used D2 because it seemed to make more sense.

Reply:
I'll do another test to find out, it was some time ago and better not to put misleading comments...

Reply:
It works without the `env:` override: https://travis-ci.org/leandro-lucarella-sociomantic/ocean/builds/299741606

But I think now I remember why it was like that. I started based on a branch that has `DIST` moved as a global. Maybe that works too though, I'll check.

Reply:
It seems to work too: https://travis-ci.org/leandro-lucarella-sociomantic/ocean/builds/299743144

I'll update.